src/test/regress/expected/qp_misc_jiras_optimizer.out

```diff
@@ -3153,12 +3153,12 @@ NOTICE:  CREATE TABLE will create partition "tbl8258_1_prt_4" for table "tbl8258
 NOTICE:  CREATE TABLE will create partition "tbl8258_1_prt_5" for table "tbl8258"
 NOTICE:  CREATE TABLE will create partition "tbl8258_1_prt_p1" for table "tbl8258"
 select pg_get_partition_def('qp_misc_jiras.tbl8258'::regclass, true);
-                                       pg_get_partition_def                                       
---------------------------------------------------------------------------------------------------
- PARTITION BY RANGE(b)                                                                           +
-           (                                                                                     +
-           START (1::double precision) END (100::double precision) EVERY (20::double precision), +
-           PARTITION p1 START (100::double precision) END (150::double precision)                +
+                                         pg_get_partition_def                                         
+------------------------------------------------------------------------------------------------------
+ PARTITION BY RANGE(b)                                                                               +
+           (                                                                                         +
+           START ('1'::double precision) END ('100'::double precision) EVERY (20::double precision), +
```

Comment:
20 is of type int4, while all others for some reason are float8 (should investigate...)

Reply:
Can we open this PR for the review anyway? (the question seems to quite complicated)

Reply:
I don't think that this research may blocks review process.

Reply:
Well, long story short, the problem seems to be in the fact: partStart and partEnd are coerced from FUNCEXPR 
> ({FUNCEXPR :funcid 316 :funcresulttype 701 :funcretset false :funcvariadic false :funcformat 1 :funccollid 0 :inputcollid 0 :args ({CONST :consttype 23 :consttypmod -1 :constcollid 0 :constlen 4 :constbyval true :constisnull false :location 149 :constvalue 4 [ 20 0 0 0 0 0 0 0 ]}) 

(and is then written to the part rule as a CONST) in unconditional way
https://github.com/arenadata/gpdb/blob/c8da24eb2d743d5058f733d3f36c817947256ade/src/backend/parser/parse_partition.c#L2416-L2433

while partEvery - only in case the [returned](https://github.com/arenadata/gpdb/blob/ADBDEV-2557/src/backend/parser/parse_partition.c#L2459) right operand's type is not equal to the original partEvery's type 
https://github.com/arenadata/gpdb/blob/5ea2818c806f9fcca61d98879304df35b5af5fb3/src/backend/parser/parse_partition.c#L2462-L2475 (otherwise it is written to the rule as is)

I have created a [separate branch](https://github.com/arenadata/gpdb/pull/311#partial-pull-merging) for the experiments aimed to fix this behavior if possible. I will also add a more extended explanation to the ticket